plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseTypeMapping.java

```diff
@@ -115,35 +115,115 @@ protected QueryRunner createQueryRunner()
     }
 
     @Test
-    public void testBasicTypes()
+    public void testTinyint()
     {
         SqlDataTypeTest.create()
-                .addRoundTrip("bigint", "123456789012", BIGINT, "123456789012")
-                .addRoundTrip("integer", "1234567890", INTEGER, "1234567890")
-                .addRoundTrip("smallint", "32456", SMALLINT, "SMALLINT '32456'")
+                .addRoundTrip("tinyint", "-128", TINYINT, "TINYINT '-128'") // min value in ClickHouse and Trino
                 .addRoundTrip("tinyint", "5", TINYINT, "TINYINT '5'")
-                .addRoundTrip("double", "123.45", DOUBLE, "DOUBLE '123.45'")
-                .addRoundTrip("real", "123.45", REAL, "REAL '123.45'")
+                .addRoundTrip("tinyint", "127", TINYINT, "TINYINT '127'") // max value in ClickHouse and Trino
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_tinyint"))
+                .execute(getQueryRunner(), trinoCreateAndInsert("test_tinyint"))
 
-                .execute(getQueryRunner(), clickhouseCreateAndInsert("tpch.test_basic_types"))
+                .addRoundTrip("Nullable(tinyint)", "NULL", TINYINT, "CAST(NULL AS TINYINT)")
+                .execute(getQueryRunner(), clickhouseCreateAndInsert("tpch.test_tinyint"));
+    }
 
-                .addRoundTrip("bigint", "NULL", BIGINT, "CAST(NULL AS BIGINT)")
-                .addRoundTrip("integer", "NULL", INTEGER, "CAST(NULL AS INTEGER)")
-                .addRoundTrip("smallint", "NULL", SMALLINT, "CAST(NULL AS SMALLINT)")
-                .addRoundTrip("tinyint", "NULL", TINYINT, "CAST(NULL AS TINYINT)")
-                .addRoundTrip("double", "NULL", DOUBLE, "CAST(NULL AS DOUBLE)")
-                .addRoundTrip("real", "NULL", REAL, "CAST(NULL AS REAL)")
+    @Test
+    public void testUnsupportedTinyint()
+    {
+        try (TestTable table = new TestTable(clickhouseServer::execute, "tpch.test_unsupported_tinyint", "(id tinyint, data tinyint) ENGINE=Log")) {
+            String tableName = table.getName();
+
+            clickhouseServer.execute(format("INSERT INTO %s VALUES (%d, %d)", tableName, 1, -129)); // min - 1
+            assertQuery(format("SELECT data FROM %s WHERE id = 1", tableName), "VALUES (127)"); // min - 1 -> max
 
-                .execute(getQueryRunner(), trinoCreateAsSelect("test_basic_types"));
+            clickhouseServer.execute(format("INSERT INTO %s VALUES (%d, %d)", tableName, 2, 128)); // max + 1
+            assertQuery(format("SELECT data FROM %s WHERE id = 2", tableName), "VALUES (-128)"); // max + 1 -> min
+        }
```

Comment:
I would change like below - simplify the code and add comment to clarify this is't the expected behavior.
```java
    @Test
    public void testUnsupportedTinyint()
    {
        // ClickHouse stores incorrect results when the values are out of supported range. This test should be fixed when ClickHouse changes the behavior.
        try (TestTable table = new TestTable(clickhouseServer::execute, "tpch.test_unsupported_tinyint", "(min tinyint, max tinyint) ENGINE=Log", ImmutableList.of("-129, 128"))) {
            assertQuery(format("SELECT min FROM %s", table.getName()), "VALUES (127)");
            assertQuery(format("SELECT max FROM %s", table.getName()), "VALUES (-128)");
        }
    }
```

Subsequent tests too.